### PR TITLE
feat: add zap support

### PR DESCRIPTION
**Pull request by ttys3**
add https://github.com/uber-go/zap support

### CODE REVIEW

**Comment by timonwong** [comment]
Hi, since #12, your pr seems need to rebase.

File: testdata/src/a/a.go [hunk 1]
```diff
@@ -49,3 +49,11 @@ func ExampleKlog() {
 	logger := klog.NewKlogr()
 	logger.Info("message", "key1") // want `odd number of arguments passed as key-value pairs for logging`
 }
+
+func ExampleZap() {
+	logger, _ := zap.NewProduction()
+	defer logger.Sync()
+	sugar := logger.Sugar()
+	sugar.Errorw("failed to fetch URL", "url", "https://example.com", "attempt", 3, "backoff", "1s")
+	sugar.With("foo", "bar").Infow("this is a info message", "key1", "value1", "key2", "value2")
```
Comment (timonwong) [inline testdata/src/a/a.go:58]: You should add a "fail" case which tests whether the linter is actually working (`// want: xxxxxx`)

**Review by timonwong — CHANGES_REQUESTED** [review]

File: logrlint.go [hunk 2]
```diff
@@ -26,6 +26,7 @@ func newAnalyzer() *analysis.Analyzer {
 var validLoggerToFuncNames = map[string]stringSet{
 	"github.com/go-logr/logr": newStringSet([]string{"Error", "Info", "WithValues"}),
 	"k8s.io/klog/v2":          newStringSet([]string{"InfoS", "InfoSDepth", "ErrorS"}),
+	"go.uber.org/zap":         newStringSet([]string{"Fatalw", "Panicw", "DPanicw", "Errorw", "Warnw", "Infow", "Debugw", "With"}),
```
Comment (timonwong) [inline logrlint.go:29]: There are two `With`, one is https://pkg.go.dev/go.uber.org/zap#Logger.With.

It will cause false-positive for this case

**Review by timonwong — CHANGES_REQUESTED** [review]

**Comment by ttys3** [comment]
OK, I'll fix this later

File: checker.go [hunk 3]
```diff
@@ -26,6 +26,19 @@ var loggerCheckersByName = loggerCheckerMap{
 			"(k8s.io/klog/v2.Verbose).ErrorS",
 		),
 	},
+	"zap": {
+		packageImport: "go.uber.org/zap",
+		funcs: newStringSet(
+			"(*go.uber.org/zap.SugaredLogger).With",
```
Comment (timonwong) [inline checker.go:32]: For receivers, there is not need to add `*`, which is invalid

**Review by timonwong — CHANGES_REQUESTED** [review]

File: checker.go (hunk 3 above)
Comment (timonwong) [reply checker.go:32]: You can run `make test` prior submitting

Comment (ttys3) [reply checker.go:32]: OK, I'll check this later

Comment (timonwong) [reply checker.go:32]: @ttys3 your updated checks looks good at a glance. I'll deep into it later

File: logrlint.go [hunk 4]
```diff
@@ -79,9 +79,15 @@ func decorateVendoredFuncs(entryFuncs stringSet, currentPkgImport, packageImport
...
 
-			leftOver := strings.TrimPrefix(importOrReceiver, "("+packageImport+".")
-			importOrReceiver = fmt.Sprintf("(%s.%s", currentPkgImport, leftOver)
+			if fn[1] == '*' {
+				pointerChar = "*"
+			}
+			prefix := "(" + pointerChar + packageImport + "."
+
+			leftOver := strings.TrimPrefix(importOrReceiver, prefix)
+			importOrReceiver = fmt.Sprintf("(%s%s.%s", pointerChar, currentPkgImport, leftOver)
```
Comment (timonwong) [inline logrlint.go:90]:
```go
			var pointerIndicator string
			if strings.HasPrefix(importOrReceiver[1:], "*") { // pointer type
				pointerIndicator = "*"
			}
			leftOver := strings.TrimPrefix(importOrReceiver, "("+packageImport+".")
			importOrReceiver = fmt.Sprintf("(%s%s.%s", pointerIndicator, currentPkgImport, leftOver)
```

**Review by timonwong — CHANGES_REQUESTED** [review]

File: logrlint.go (hunk 4 above)
Comment (timonwong) [reply logrlint.go:90]: And your change to extract `prefix` alone is good. So it's just a bit rename

**Comment by timonwong** [comment]
Please note that by default, only logr and klog is enabled right now (latest base)

**Comment by ttys3** [comment]
Should we enable zap by default ?
`zap` logger seem popular enough, and is widely used

File: logrlint.go (hunk 4 above)
Comment (ttys3) [reply logrlint.go]: done

**Comment by ttys3** [comment]
rebased to master

**Comment by ttys3** [comment]
I see. it is just doc change. I've updated README.md

**Review by timonwong — APPROVED** [review]
LGTM

**Pull request merged**